drivers/dma/xilinx/xilinx_dma.c

```diff
@@ -760,15 +754,28 @@ static void xilinx_dma_complete_descriptor(struct xilinx_dma_chan *chan)
 {
 	struct xilinx_dma_tx_descriptor *desc, *next;
 
-	if (list_empty(&chan->active_list))
-		return;
 
 	list_for_each_entry_safe(desc, next, &chan->active_list, node) {
+
+		/* Check whether the last segment in this descriptor has been completed. */
+		const struct xilinx_dma_tx_segment * const tail_segment =
+			list_last_entry(&desc->segments, struct xilinx_dma_tx_segment, node);
+
+		if ( ! (tail_segment->hw.status & XILINX_DMA_BD_CMPLT) )
+			break;  // we've processed all the completed descriptors so far
```

Comment:
This code loops over "active" descriptors until "we" (the driver) have processed all completed descriptors; as soon as we find a descriptor not completed yet, we break. So the comment seems OK to me.